packages/core/src/common/types.ts

```diff
@@ -31,6 +31,11 @@ export interface Prioritizeable<T> {
     readonly priority: number;
     readonly value: T;
 }
+
+export function isNumber(obj: unknown): obj is number {
+    return (typeof obj === 'number' && !isNaN(obj));
+}
```

Comment:
I do not think the following is necessary, you can use `Number.isFinite()` to achieve the same thing (and it handles `Infinity` and `-Infinity`).

```bash
> Number.isFinite(NaN)
false
> Number.isFinite(Infinity)
false
> Number.isFinite(-Infinity)
false
> Number.isFinite(10)
true
> Number.isFinite("")
false
> Number.isFinite("10")
false
```

I do not think we need to copy vscode for such a method.